### DISCUSSION TITLE
Improvements

### DISCUSSION THREAD

**Discussion by oggioniale** [ropensci/ReLTER#90] [Ideas]
Some improvements:

- ask to DEIMS-SDR developer the API for requests
- ...

**Comment by micha-silver** [comment]
- We currently have a repo of almost 300MB. (!)
About 210MB are images in the both the ./man and .docs directories. Some of these images are 15-25 MB each. Do we really need such high quality images?

- Following this [rspatial blog post](https://r-spatial.org/r/2022/04/12/evolution.html) we should get organized to remove dependencies on sp. raster, and rgdal. Functions using these should be changed to use instead sf and terra.
- Currently we use two different packages for displaying maps: leaflet and tmap. The leaflet package is used in 7 functions, and tmap in only two. In an effort to reduce package dependencies, we should focus on only one of these.

- Last year I added tmap since I needed to display raster images (i.e. in the get_site_ODS() function), and leaflet is a bit troublesome and tricky with raster data. However now there is a new function in terra (which we need installed anyway) called plet (yes, that's the spelling :-) ) which displays a terra::SpatRaster with leaflet. This function requires leaflet >  2.1.1. But that version so far is only available from github. That means installing from github, not CRAN. Can we do that?
So I suggest to change the functions that currently use tmap over to leaflet. And remove the dependency on tmap.

- If we push up the minimum R version to 4.1, then we can convert all pipes ('%>%') to the new, builtin '|>' and then remove dependency on magrittr. 

-  There's need for a new function to find DEIMS URL for a **network**. (Perhaps inside the `get_ilter_generalinfo()` function?) . This should be coordinated with DEIMS, so that it can be done with an API call, similar to retrieving the URL of a site.

**Reply by oggioniale** [reply]
- Remove most of the imports to the packages: if the function with specific import are execute, check if the dependance is already installed, otherwise ask to install.

**Reply by micha-silver** [reply]
I think we need to do two things:

1. Packages that are used only rarely, we should move to "Suggests" and then put into the code a test if the package is available, and send a message to user if not.
2. In any case, try to remove as many packages as possible.

**Comment by stopopol** [comment]
Please keep in mind that the dataset functionality of DEIMS will be transferred to the DAR in the future. The dataset functions of ReLTER will therefore have to be adjusted. However, there is no timetable for neither the official release of the DAR nor the time of the dataset migration from DEIMS to the DAR (yet).
Christoph
